src/main/kotlin/com/whatever/domain/user/model/User.kt

```diff
@@ -83,13 +88,29 @@ class User(
         userStatus = newStatus
     }
 
+    fun updateBirthDate(
+        newBirthDate: LocalDate,
+        userZoneId: ZoneId = KST_ZONE_ID,
+    ) {
+        val todayInUserZone = DateTimeUtil.zonedNow(userZoneId).toLocalDate()
+        if (newBirthDate.isAfter(todayInUserZone)) {
+            throw UserIllegalArgumentException(
+                errorCode = INVALID_BIRTH_DATE,
+                errorUi = ErrorUi.Toast("미래의 날짜를 생일로 설정할 수 없어요.")
+            )
+        }
```

Comment:
```suggestion
    require(newBirthDate.isBefore(todayInUserZone)) {
            throw UserIllegalArgumentException(
                errorCode = INVALID_BIRTH_DATE,
                errorUi = ErrorUi.Toast("미래의 날짜를 생일로 설정할 수 없어요.")
            )
        }
```
이건 갠취의 영역인데용, require나 check도 코틀린 스러운 개발 + 명확한 의도가 보여서 좋아요 
저희는 물론 커스텀 익셉션을 쓰긴 하지만, 위 처럼도 가능하긴 하다,, 정도? 
그냥 참고용으로 보셔도 조아요~

![image](https://github.com/user-attachments/assets/f43cd60b-95e2-4a0b-aa6f-bc428d8e59b7)

Reply:
https://kotlinlang.org/docs/exceptions.html#throw-exceptions-with-precondition-functions 참고한건 여기 !